pkg/bindings/volumes/volumes.go

```diff
@@ -75,15 +75,15 @@ func List(ctx context.Context, filters map[string][]string) ([]*entities.VolumeL
 }
 
 // Prune removes unused volumes from the local filesystem.
-func Prune(ctx context.Context) ([]*entities.VolumePruneReport, error) {
+func Prune(ctx context.Context, opts entities.VolumePruneOptions) ([]*entities.VolumePruneReport, error) {
 	var (
 		pruned []*entities.VolumePruneReport
 	)
 	conn, err := bindings.GetClient(ctx)
 	if err != nil {
 		return nil, err
 	}
-	response, err := conn.DoRequest(nil, http.MethodPost, "/volumes/prune", nil, nil)
+	response, err := conn.DoRequest(nil, http.MethodPost, "/volumes/prune", opts.Filters, nil)
```

Comment:
This is not correct. You have to add the filters to a new params map. Otherwise you will lose the `filters` key
```
params := url.Values{}
if opts.Filters != nil {
	filterString, err := bindings.FiltersToString(opts.Filters)
	if err != nil {
		return nil, err
	}
	params.Set("filters", filterString)
}
response, err := conn.DoRequest(nil, http.MethodPost, "/volumes/prune", params, nil)
```

Reply:
:+1: Thanks for the feedback. How would I have found this error on my own? I tried to update the tests in `test/apiv2/30-volumes.at` and they were passing. Those tests had the params map with filter key encoded in the url manually so I'm guessing thats why they worked. Was there another test(s) I could have ran on my dev box? I ran the following things to test on my machine

1. Manual `bin/podman` commands to test functionality
2. `sudo ./test-apiv2 30-volumes`
3. `go test -v test/e2e/config.go test/e2e/config_amd64.go test/e2e/common_test.go test/e2e/libpod_suite_test.go test/e2e/volume_prune_test.go`
4. `make lint` in the gate container

(you can see my test output in the [issue](https://github.com/containers/podman/issues/8672) for this PR)

Reply:
Our CI would have catched  this. The test in `test/e2e/volume_prune_test.go` has to be run twice, for `podman` and `podman-remote`. This code path is only taken with `podman-remote`.

You can run the remote integration tests with `make remoteintegration` but make sure to add `FIt` instead of `It` in front of  your test otherwise the full test suite will be run but don't forget to remove the `FIt` before you commit your changes.

Reply:
Fixed: https://github.com/containers/podman/pull/8689/commits/0a39fdaa2abf0c5905d6665f42ce74ffe4401093